config_parser.cc

```diff
@@ -148,6 +148,9 @@ NginxConfigParser::TokenType NginxConfigParser::ParseToken(std::istream* input,
 }
 
 bool NginxConfigParser::Parse(std::istream* config_file, NginxConfig* config) {
+  int start_block_counter=0;
+  int end_block_counter=0;
```

Comment:
You could combine these into one counter, like "start blocks - end blocks", and check when it's zero instead of if two counters are equal.